libbeat/common/docker/watcher.go

```diff
@@ -289,17 +296,32 @@ func (w *watcher) watch() {
 				time.Sleep(1 * time.Second)
 				break WATCH
 
+			case <-tickChan.C:
+				if time.Since(time.Unix(w.lastValidTimestamp, 0)) > 10*time.Minute {
+					logp.Info("No events received withing 10 minutes, restarting watch call")
+					w.lastValidTimestamp = time.Now().Unix()
```

Comment:
why updating this here? If watch call has gone mad maybe we missed some previous events?

Reply:
I changed code to use separate var for tracking watch activity